src/network/createAsyncComparison.ts

```diff
@@ -46,7 +46,11 @@ export default async function createAsyncComparison(
     fallbackShas,
   }: EnvironmentResult,
   logger: Logger,
-): Promise<CreateAsyncComparisonResult> {
+): Promise<CreateAsyncComparisonResult | null> {
+  if (beforeSha === afterSha) {
+    return null;
```

Comment:
I am wondering if we should throw here instead, so we don't end up hiding a env resolution bug on accident.

Also, if beforeSha is undefined, this won't be right either.